viewing_party/party.py

```diff
@@ -1,23 +1,286 @@
+    
+    Parameters: 
+        user_data: dict, 
+            {"watched": [{}, {}, {}]}
+        movie: dict
+            {
+                "title": "Title A",
+                "genre": "Horror",
+                "rating": 3.5
+            }
+
+    Returns:
+        user_data
+    """
+    
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    """Add the movie to the "watchlist" list inside of user_data
+    
+    Parameters: 
+        user_data: dict, 
+            {"watchlist": [{}, {}, {}]}
+        movie: dict
+            {
+                "title": "Title A",
+                "genre": "Horror",
+                "rating": 3.5
+            }
+
+    Returns:
+        user_data
+    """
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, title):
+    """Move a movie from the "watchlist" list to the "watched" list inside of user_data.
+    
+    Parameters: 
+        user_data: dict, 
+            {"watchlist": [{}, {}, {}], "watched": [{}, {}, {}]}
+        title: str
+            The title of the movie the user has watched
+
+    Returns:
+        user_data
+    """
+    for movie in user_data["watchlist"]:
+        if movie.get("title") == title:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie) 
+            return user_data
+    return user_data
+
+
+def get_watched_avg_rating(user_data):
+    """Calculate the average rating of the movies in the user's watched list.
+
+    Parameters:
+        user_data (dict): A dictionary containing user information.
+            Expected to have a key "watched", whose value is a list of 
+            movie dictionaries. Each movie dictionary must contain a 
+            "rating" key with a numeric value.
+
+    Returns:
+        float: The average rating of all movies in the "watched" list.
+            Returns 0.0 if the list is empty.
+    """
+    num_of_movies = len(user_data["watched"])
+    rating_total = 0
+    if not user_data["watched"]:
+        return 0.0
+    for movie in user_data["watched"]:
+        rating_total += movie["rating"]
+    rating_avg = rating_total / num_of_movies
+    return rating_avg
+
+
+def get_most_watched_genre(user_data):
+    """Determine the most frequently watched genre for a user.
+
+    Parameters:
+        user_data (dict): A dictionary containing user information.
+            Expected to have a key "watched", whose value is a list of 
+            movie dictionaries. Each movie dictionary must contain 
+            a "genre" key with a string value.
+
+    Returns:
+        str: The genre that appears most frequently in the "watched" list.
+        None: If the "watched" list is empty.
+    """
+
+    frequently_watched_genre = {}
+    max_num = 0
+    most_watched_genre = ""
+    if not user_data["watched"]:
+        return None
+    for movie in user_data["watched"]:

```

Comment:
Consider adding some blank lines to logically and visually separate this code so it's easier to read.

```suggestion
    frequently_watched_genre = {}
    max_num = 0
    most_watched_genre = ""
    
    if not user_data["watched"]:
        return None
        
    for movie in user_data["watched"]:

```